### DISCUSSION TITLE
Livewire won't update tab title after a new refresh

### DISCUSSION THREAD

**Discussion by aigorlaxy** [livewire/livewire#8049] [1. Help]
`` public function render()
    {
        $item = $this->item;
        $search = $this->search;
        $itemsPerPage = $this->itemsPerPage;
        $title = null;
        switch ($item) {
            case 'order':
                $title = 'Pedidos';
                break;
            case 'soldProduct':
                $title = 'Produtos Vendidos';
                break;
            case 'product':
                $title = 'Produtos';
                break;
            case 'component':
                $title = 'Componentes';
                break;
            case 'printing':
                $title = 'Impressões';
                break;
            case 'rawMaterial':
                $title = 'Matéria Prima';
                break;
            case 'plainTshirt':
                $title = 'Camiseta Lisa';
                break;
        }
        $stocks = StockModel::with($item)
            ->whereHas($item, function ($query) use ($search) {
                $query->where('label', 'like', "%$search%");
            })
            ->paginate($itemsPerPage);

        return view('livewire.stock')->with([
            'stocks' => $stocks
        ])
            ->title("Estoque $title");
    }``
    
    This is my code. It should update $title after every refresh. In the first refresh everything works fine, but in the subsequente, the title stays as the last one.

Any idea why? Thanks!

**Comment by lrljoe** [comment]
Please format your code properly, I started looking at it but the lack of formatting means kt isn't easy.

Why are you converting:

```
$title = $this->title;
```

You say it's the previous value, can you please share your whole component, in a formatted way, so wrap in three ` marks please!

Also, look at withWhereHas rather than your approach as it's a little cleaner!

And try not to put all of that in your render method, as it's less than ideal, as it will all be executed on every lifecycle, when it may not need to be.

Seeing your whole component, and the associated blade will help everyone to help you.
